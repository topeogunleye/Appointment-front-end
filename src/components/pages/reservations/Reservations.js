import React from 'react';

function Reservations() {
  return (
    <div>This is the page that shows all the reservations you have created as a user</div>
  );
}

export default Reservations;
