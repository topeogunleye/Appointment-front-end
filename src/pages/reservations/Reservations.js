import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Sidebar from '../../components/Sidebar/Sidebar';
import { fetchServices } from '../../redux/reservationSlice';

const Reservations = () => {
  const dispatch = useDispatch();
  const reservations = useSelector((state) => state.reservation.formData);
  console.log(reservations);
  const isLoading = useSelector((state) => state.isLoading);
  const user = useSelector((state) => state.auth.user); // assuming you have a currentUser state
  console.log(user);
  useEffect(() => {
    dispatch(fetchServices());
  }, [dispatch]);

  const userReservations = reservations.filter(
    (reservation) => reservation.id === user.id,
  );

  return (
    <div className="grid grid-cols-custom">
      <Sidebar />
      <div className="container mx-auto my-4">
        <h1 className="text-3xl font-bold mb-4">Reservations</h1>
        {isLoading ? (
          <p>Loading...</p>
        ) : (
          <table className="min-w-full divide-y divide-gray-200">
            <thead className="bg-gray-50">
              <tr>
                <th
                  scope="col"
                  className="px-6 py-3 text-xs font-medium text-gray-500 uppercase tracking-wider"
                >
                  Vehicle
                </th>
                <th
                  scope="col"
                  className="px-6 py-3 text-xs font-medium text-gray-500 uppercase tracking-wider"
                >
                  Model
                </th>
                <th
                  scope="col"
                  className="px-6 py-3 text-xs font-medium text-gray-500 uppercase tracking-wider"
                >
                  Year
                </th>
                <th
                  scope="col"
                  className="px-6 py-3 text-xs font-medium text-gray-500 uppercase tracking-wider"
                >
                  Color
                </th>
                <th
                  scope="col"
                  className="px-6 py-3 text-xs font-medium text-gray-500 uppercase tracking-wider"
                >
                  Location
                </th>
                <th
                  scope="col"
                  className="px-6 py-3 text-xs font-medium text-gray-500 uppercase tracking-wider"
                >
                  Service
                </th>
              </tr>
            </thead>
            <tbody className="bg-white divide-y divide-gray-200">
              {userReservations.map((service) => (
                <tr key={service.id}>
                  <td className="px-6 py-4 whitespace-nowrap text-sm font-medium text-gray-900">
                    {service.vehicle}
                  </td>
                  <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">
                    {service.model}
                  </td>
                  <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">
                    {service.year}
                  </td>
                  <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">
                    {service.color}
                  </td>
                  <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">
                    {service.location}
                  </td>
                  <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">
                    {service.service}
                  </td>
                </tr>
              ))}
            </tbody>
          </table>
        )}
      </div>
    </div>
  );
};

export default Reservations;
