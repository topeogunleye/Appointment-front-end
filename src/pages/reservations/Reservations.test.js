import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Reservations from './Reservations';
import store from '../../redux/configureStore'; // import your Redux store

test('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Reservations />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
