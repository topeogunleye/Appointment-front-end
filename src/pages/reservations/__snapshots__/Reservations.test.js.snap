// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="min-h-screen flex items-center justify-center bg-gray-100"
>
  <button
    className="absolute top-1 left-1 sm:top-4 sm:left-4 hover:bg-indigo-600 text-white bg-green py-1 px-1 sm:py-2 sm:px-4"
    onClick={[Function]}
    type="button"
  >
    Â« Go Back
  </button>
  <div
    className="max-w-md w-full"
  >
    <div
      className="text-center text-2xl font-bold mb-8"
    >
      <h1
        className="text-gray-900"
      >
        Welcome!
      </h1>
    </div>
    <div
      className="bg-white shadow-md rounded px-8 pt-6 pb-8 mb-4"
    >
      <p
        className="text-center text-gray-700 text-base"
      >
        You have no reservations yet.
      </p>
    </div>
  </div>
</div>
`;
